packages/block-library/src/navigation-menu-item/edit.js

```diff
@@ -1,93 +1,81 @@
-/**
- * External dependencies
- */
-import { invoke } from 'lodash';
-
 /**
  * WordPress dependencies
  */
 import { withSelect } from '@wordpress/data';
 import {
-	Dropdown,
 	ExternalLink,
-	IconButton,
 	PanelBody,
 	TextareaControl,
 	TextControl,
+	Toolbar,
 	ToggleControl,
+	ToolbarButton,
 } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 import {
+	BlockControls,
 	InnerBlocks,
 	InspectorControls,
 	PlainText,
+	URLPopover,
 } from '@wordpress/block-editor';
 import {
 	Fragment,
-	useCallback,
 	useRef,
+	useState,
 } from '@wordpress/element';
 
-/**
- * Internal dependencies
- */
-import MenuItemActions from './menu-item-actions';
-const POPOVER_PROPS = { noArrow: true };
-
 function NavigationMenuItemEdit( {
 	attributes,
-	clientId,
 	isSelected,
 	isParentOfSelectedBlock,
 	setAttributes,
 } ) {
 	const plainTextRef = useRef( null );
-	const onEditLableClicked = useCallback(
-		( onClose ) => () => {
-			onClose();
-			invoke( plainTextRef, [ 'current', 'textarea', 'focus' ] );
-		},
-		[ plainTextRef ]
-	);
+	const [ isLinkOpen, setLinkOpen ] = useState( false );
+	const { label, url } = attributes;
 	let content;
 	if ( isSelected ) {
 		content = (
 			<div className="wp-block-navigation-menu-item__edit-container">
 				<PlainText
 					ref={ plainTextRef }
 					className="wp-block-navigation-menu-item__field"
-					value={ attributes.label }
-					onChange={ ( label ) => setAttributes( { label } ) }
+					value={ label }
+					onChange={ ( labelValue ) => setAttributes( { label: labelValue } ) }
```

Comment:
Because I also need to get the url from `attributes`, it's better to destructure all its properties. Also more consistent with how we're doing it elsewhere. But then the linter complains if we use `label` in `onChange` because it's already declared in the upper scope, so I had to change that too.

Reply:
@draganescu: Our ESLint configuration doesn't allow shadowing. The intention is that this helps with reducing ambiguity. 

```js
const foo = 123;
const bar = ( foo ) => { // Error: `foo` is already in scope here
	return 'hello ' + foo;
};
```